echopop/inversion/inversion_matrix.py

```diff
@@ -0,0 +1,1230 @@
+    **kwargs : dict
+        Additional keyword arguments
+
+    Returns
+    -------
+    list
+        List of configured lmfit.Minimizer objects for Monte Carlo optimization
+
+    Notes
+    -----
+    The function generates Monte Carlo parameter realizations and creates
+    a separate Minimizer for each realization. Each Minimizer is configured
+    with the same objective function but different parameter starting points.
+
+    The objective function minimizes the sum of squared differences between
+    predicted and measured Sv values across all valid frequencies.
+    """
+    # Generate parameter sets
+    parameter_sets = MCInvParameters(
+        scattering_params, simulation_settings["mc_realizations"], simulation_settings["rng"]
+    )
+
+    # Generate `lmfit.Parameters` objects for each realization
+    parameters_lmfit = parameter_sets.to_lmfit_samples()
+
+    # Find which values thresholded out
+    valid_idx = np.argwhere(Sv_measured > -999.0).flatten()
+
+    # Get the valid center frequencies
+    center_frequencies = np.array(Sv_measured.index.values, dtype=float)[valid_idx]
+
+    # Turn into a numpy array
+    Sv_measured = Sv_measured.to_numpy()[valid_idx].astype(float)
+
+    # Initialize with Monte Carlo sampling
+    if simulation_settings["monte_carlo"]:
+        parameters_lmfit = monte_carlo_initialize(
+            parameters_lmfit, center_frequencies, Sv_measured, parameter_sets, model_settings
+        )
+    else:
+        parameters_lmfit = parameters_lmfit[0]
+
+    # Set `iter_cb` if verbose
+    if verbose:
+        if "iter_cb" in simulation_settings and simulation_settings["iter_cb"]:
+            iter_cb_arg = simulation_settings["iter_cb"]
+        else:
+            iter_cb_arg = echopop_optim_cb
+    else:
+        iter_cb_arg = None
+    # Generate `Minimizer` function class required for bounded optimization
+    return [
+        Minimizer(
+            fit_Sv,
+            parameters_lmfit,
+            fcn_args=(
+                Sv_measured,
+                center_frequencies,
+                parameter_sets,
+                model_settings,
+            ),
+            iter_cb=iter_cb_arg,
+            nan_policy="omit",
+        )
+    ]
+
+
+def fit_Sv(
+    parameters, Sv_measured, center_frequencies, parameters_meta: InvParameters, model_settings
+):
+
+    # Extract the `lmfit.Parameters` values
+    parameter_set = parameters.valuesdict()
+
+    # Rescale to the original scale if parameters were normalized
+    if parameters_meta.is_scaled:
+        parameter_set = parameters_meta.unscale_dict(parameter_set)
+
+    # Compute Sv
+    Sv_prediction = model_settings["model_function"](
+        center_frequencies=center_frequencies,
+        **parameter_set,
+        **model_settings["environment"],
+        **model_settings,
+    )
+
+    return objective(Sv_prediction, Sv_measured)
+
+
+def objective(Sv_prediction: np.ndarray[float], Sv_measured: np.ndarray[float]):
```

Comment:
I suggest absorbing `objective` into `fit_Sv` so that there is one less thing to keep track of. The minimizer is functioning on `fit_Sv` so it would be fine to have that in there. In fact, i'd even suggest renaming it to be something like `cost_Sv` or `objective_Sv_diff`